lisa/tools/lsvmbus.py

```diff
@@ -0,0 +1,133 @@
+        file_path = wget_tool.get(self.__lsvmbus_path, "$HOME/.local/bin")
+        # make the download file executable
+        self.node.execute(f"chmod +x {file_path}")
+        self._command = file_path
+        return True
+
+    def get_vmbuses(self, force_run: bool = False) -> List[VmBus]:
+        cached_result = None
+        if (not self._vmbuses) or force_run:
+            cached_result = self.run("-vv")
+            if cached_result.exit_code != 0:
+                raise LisaException(
+                    f"get unexpected non-zero exit code {cached_result.exit_code} "
+                    f"when run {self.command} -vv."
+                )
+            # segment output of lsvmbus -vv
+            # VMBUS ID  1: Class_ID = {525074dc-8985-46e2-8057-a307dc18a502}
+            # - [Dynamic Memory]
+            # \r\n\t
+            # Device_ID = {1eccfd72-4b41-45ef-b73a-4a6e44c12924}
+            # \r\n\t
+            # Sysfs path: /sys/bus/vmbus/devices/1eccfd72-4b41-45ef-b73a-4a6e44c12924
+            # \r\n\t
+            # Rel_ID=1, target_cpu=0
+            # \r\n\r\n
+            # VMBUS ID  2: Class_ID = {32412632-86cb-44a2-9b5c-50d1417354f5}
+            # - Synthetic IDE Controller\r\n\t
+            # Device_ID = {00000000-0000-8899-0000-000000000000}
+            # \r\n\t
+            # Sysfs path: /sys/bus/vmbus/devices/00000000-0000-8899-0000-000000000000
+            # \r\n\t
+            # Rel_ID=2, target_cpu=0
+            # \r\n\r\n
+            # VMBUS ID  3: Class_ID = {32412632-86cb-44a2-9b5c-50d1417354f5}
+            # - Synthetic IDE Controller
+            # \r\n\t
+            # Device_ID = {00000000-0001-8899-0000-000000000000}
+            # \r\n\t
+            # Sysfs path: /sys/bus/vmbus/devices/00000000-0001-8899-0000-000000000000
+            # \r\n\t
+            raw_array = [s.replace("\t", "") for s in cached_result.stdout.splitlines()]
+            sub_array = []
+            channel_vp_map_list: List[ChannelVPMap] = []
+            remove_bracket_pattern = re.compile(r"\{(.*?)\}")
+
+            for value in raw_array:
+                if value:
+                    sub_array.append(value)
+                else:
+                    if sub_array:
+                        # VMBUS ID  1: Class_ID = {525074dc-8985-46e2-8057-a307dc18a502}
+                        # - [Dynamic Memory]
+                        # => Dynamic Memory
+                        vmbus_name = (
+                            sub_array[0]
+                            .split("-")[-1]
+                            .strip()
+                            .replace("[", "")
+                            .replace("]", "")
+                        )
+                        # VMBUS ID  1: Class_ID = {525074dc-8985-46e2-8057-a307dc18a502}
+                        # - [Dynamic Memory]
+                        # => 525074dc-8985-46e2-8057-a307dc18a502
+                        class_id = remove_bracket_pattern.findall(sub_array[0])
```

Comment:
Those logic can be in `VmBus` or `ChannelVPMap`, so that they can self contained the format. Then call like `VmBus.parse`